composer/utils/checkpoint.py

```diff
@@ -68,6 +68,23 @@ def _get_write_mode(name: str) -> str:
     raise ValueError(f'{name} does not end with a valid tarfile extension.')
 
 
+class PartialFilePath:
+
+    def __init__(self, filename: str, folder: Optional[str] = None):
+        self.folder = folder
+        self.filename = filename
+
+    def format(self, state: State, is_deepspeed: bool = False) -> str:
+        suffix = '.tar' if is_deepspeed and not is_tar(self.filename) else ''
```

Comment:
should we add `.pt` suffix to checkpoints?

Reply:
I dont think we should automaticaly add -- but i believe the default filenames are all .pt

Reply:
Optional nit: I'd like a comment which explains filename might already have a suffix or maybe change variable name? (Not sure if its necessary, but its a bit confusing to have a suffix be `''` since it's not clear the filename has `.pt` (potentially) already added)